### PR TITLE
feat(vitals-alert): reading project data in FE

### PR DESCRIPTION
**Pull request by scefali**
This PR adds the following logic for the vitals alert:

- Ensure we have 100 transactions of the relevant vital within the organization to show the alert
- Pick the project with the most relevant transactions if we can't select every project

stuff

### CODE REVIEW

File: static/app/components/performance/vitalsAlert/types.tsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+  appWarmStartCount: number;
+  fcpCount: number;
+  lcpCount: number;
 }
 
-export type VitalsKey = keyof VitalsResult;
+// couldn't figure out how to do this with an interface
+export type VitalsResult = BaseVitalsResult & {
+  projectData: Array<BaseVitalsResult & {projectId: string}>;
+};
```
Comment (udameli) [inline static/app/components/performance/vitalsAlert/types.tsx:16]: Can you extend the interface like this `interface VitalsResult extends BaseVitalsResult {
    projectData: ...
    }
` maybe?

Comment (scefali) [reply static/app/components/performance/vitalsAlert/types.tsx]: @udameli This didn't work originally but I fixed the issue so now I can use this style, thanks!

File: static/app/components/performance/vitalsAlert/vitalsAlertCTA.tsx [hunk 2]
```diff
@@ -70,18 +75,34 @@ export default function VitalsAlertCTA({data, dismissAlert}: Props) {
...
+      return false;
+    }
+    // must have either global views enabled or we can pick a specific project
+    return hasGlobalViews || bestProjectData;
+  };
 
-  // if worst vital is better than Sentry users, we shouldn't show this alert
-  if (sentryDiff < 0) {
+  // TODO: log experiment
+  if (!vital || !showVitalsAlert()) {
```
Comment (Neo-Zhixing) [inline static/app/components/performance/vitalsAlert/vitalsAlertCTA.tsx:105]: when !vital, showVitalsAlert will always return false right? Don't think you need this additional check here.

**Review by PixelDust22 — APPROVED** [review]

File: static/app/components/performance/vitalsAlert/vitalsAlertCTA.tsx [hunk 3]
```diff
@@ -70,18 +75,34 @@ export default function VitalsAlertCTA({data, dismissAlert}: Props) {
...
-  if (!vital) {
-    return null;
-  }
-  const ourValue = data[vital];
-  if (!ourValue) {
-    return null;
-  }
-  const sentryDiff = getRelativeDiff(ourValue, SENTRY_CUSTOMERS[vital]);
-  const industryDiff = getRelativeDiff(ourValue, INDUSTRY_STANDARDS[vital]);
+  const ourValue = vital ? data[vital] : 0;
```
Comment (Neo-Zhixing) [inline static/app/components/performance/vitalsAlert/vitalsAlertCTA.tsx:78]: nit: Can we try not to use "our" in code? Maybe "userVitalValue"?

**Review by PixelDust22 — APPROVED** [review]

File: static/app/components/performance/vitalsAlert/vitalsAlertCTA.tsx (hunk 2 above)
Comment (scefali) [reply static/app/components/performance/vitalsAlert/vitalsAlertCTA.tsx:105]: @Neo-Zhixing Correct, but TS doesn't know that. I'm doing that here so TS knows `vital` isn't falsy.

**Pull request merged**